### PR TITLE
EventCollector: bugfix + optional callback.

### PR DESCRIPTION
**Pull request by mzhurovich**
@dkorolev PTAL.

### CODE REVIEW

File: EventCollector/event_collector.h [hunk 1]
```diff
@@ -59,16 +60,19 @@ struct LogEntry {
...
   EventCollectorHTTPServer(int http_port,
-                         std::ostream& ostream,
-                         const bricks::time::MILLISECONDS_INTERVAL tick_interval_ms,
-                         const std::string& route = "/log",
-                         const std::string& response_text = "OK\n")
+                           std::ostream& ostream,
+                           const bricks::time::MILLISECONDS_INTERVAL tick_interval_ms,
+                           const std::string& route = "/log",
+                           const std::string& response_text = "OK\n",
+                           std::function<void(const LogEntry&)> callback = {})
```
Comment (dkorolev) [inline EventCollector/event_collector.h]: `= nullptr` works too.

File: EventCollector/test.cc [hunk 2]
```diff
@@ -39,10 +39,25 @@ DEFINE_int32(event_collector_test_port, 8089, "Local port to run the test.");
 
 using bricks::strings::Printf;
 
+struct EventReceiver {
+  EventReceiver() : count(0u) {}
```
Comment (dkorolev) [inline EventCollector/test.cc]: `, last_t(0ull)` ?

**Comment by dkorolev** [comment]
Minor comments, LGTM otherwise.

File: EventCollector/event_collector.h (hunk 1 above)
Comment (mzhurovich) [inline EventCollector/event_collector.h]: `{}` is shorter ;)

File: EventCollector/test.cc (hunk 2 above)
Comment (mzhurovich) [inline EventCollector/test.cc]: Fixed!

File: EventCollector/event_collector.h (hunk 1 above)
Comment (dkorolev) [inline EventCollector/event_collector.h]: As long as it doesn't create temporary objects and is as fast -- sure! ;-)

**Comment by dkorolev** [comment]
LGTM!

**Pull request merged**